server/src/main/scala/org/http4s/server/armeria/ArmeriaServerBuilder.scala

```diff
@@ -0,0 +1,334 @@
+    * @see [[ArmeriaBuilder#http(java.net.InetSocketAddress)]]
+    */
+  def withHttp(port: Int): Self = {
+    armeriaServerBuilder.http(port)
+    this
+  }
+
+  /** Adds an HTTPS port that listens on all available network interfaces.
+    *
+    * @param port the HTTPS port number.
+    * @see [[ArmeriaBuilder#https(java.net.InetSocketAddress)]]
+    */
+  def withHttps(port: Int): Self = {
+    armeriaServerBuilder.https(port)
+    this
+  }
+
+  /** Sets the [[ChannelOption]] of the server socket bound by [[BackendServer]].
+    * Note that the previously added option will be overridden if the same option is set again.
+    *
+    * @see [[https://armeria.dev/docs/advanced-production-checklist Production checklist]]
+    */
+  def withChannelOption[T](option: ChannelOption[T], value: T): Self = {
+    armeriaServerBuilder.channelOption(option, value)
+    this
+  }
+
+  /** Sets the [[ChannelOption]] of sockets accepted by [[BackendServer]].
+    * Note that the previously added option will be overridden if the same option is set again.
+    *
+    * @see [[https://armeria.dev/docs/advanced-production-checklist Production checklist]]
+    */
+  def withChildChannelOption[T](option: ChannelOption[T], value: T): Self = {
+    armeriaServerBuilder.childChannelOption(option, value)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] from the specified `keyCertChainFile`,
+    * `keyFile` and `keyPassword`.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(keyCertChainFile: File, keyFile: File, keyPassword: Option[String]): Self = {
+    armeriaServerBuilder.tls(keyCertChainFile, keyFile, keyPassword.orNull)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] with the specified `keyCertChainInputStream`,
+    * `keyInputStream` and `keyPassword`.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(
+      keyCertChainInputStream: InputStream,
+      keyInputStream: InputStream,
+      keyPassword: Option[String]): Self = {
+    armeriaServerBuilder.tls(keyCertChainInputStream, keyInputStream, keyPassword.orNull)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] with the specified cleartext [[PrivateKey]] and
+    * [[X509Certificate]] chain.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(key: PrivateKey, keyCertChain: X509Certificate*): Self = {
+    armeriaServerBuilder.tls(key, keyCertChain: _*)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] with the specified [[KeyManagerFactory]].
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(keyManagerFactory: KeyManagerFactory): Self = {
+    armeriaServerBuilder.tls(keyManagerFactory)
+    this
+  }
+
+  /** Configures SSL or TLS of the [[BackendServer]] with an auto-generated self-signed certificate.
+    * '''Note:''' You should never use this in production but only for a testing purpose.
```

Comment:
People are going to use this in production. :laughing: The core repo has been stubborn about providing this, but it's a popular feature.  I think you're probably right to provide it, and to lecture people not to abuse it.

Reply:
I would hide this from the `ArmeriaServerBuilder` and let users only configure this with `withArmeriaBuilder(customizer)`